ledger/participant-integration-api/src/main/scala/platform/apiserver/services/ApiSubmissionService.scala

```diff
@@ -268,10 +269,14 @@ private[apiserver] final class ApiSubmissionService private[services] (
 
   private def toStatus(errorCause: ErrorCause) =
     errorCause match {
-      case e: ErrorCause.DamlLf =>
-        Status.INVALID_ARGUMENT.withDescription(e.explain)
-      case e: ErrorCause.LedgerTime =>
-        Status.ABORTED.withDescription(e.explain)
+      case cause @ ErrorCause.DamlLf(error) =>
+        error match {
+          case ContractNotFound(_) | ReplayMismatch(_) =>
+            Status.ABORTED.withDescription(cause.explain)
```

Comment:
Don't we actually want to return the NOT_FOUND code?

Reply:
Not sure; I find it suspect to return `NOT_FOUND` for something that's a command, not a query.